src/domain-registry-essentials/architecture-overview.md

```diff
@@ -1,123 +1,135 @@
 
 {% hint style="info" %}
-Updates to our Resolver smart contract are incremental and non-breaking, meaning that we add new features without disrupting the work of existing domains that haven't got updated. Any Resolver smart contract must implement [IResolver interface](https://github.com/unstoppabledomains/dot-crypto/blob/master/contracts/IResolver.sol), which defines the basic set of functionality and guarantees compatibility between different implementations.
+Updates to our `Resolver` smart contract are incremental and non-breaking. All `Resolver` smart contracts must adhere to our [IResolver interface](https://github.com/unstoppabledomains/dot-crypto/blob/master/contracts/IResolver.sol). This interface defines the basic set of functionality and guarantees compatibility between different implementations.
 {% endhint %}
 
-As mentioned, every domain represents an ERC-721 non-fungible token. Every ERC-721 token is identified by a unique number, which is called `tokenId`. To make domains identifiable, we use a process called [Namehashing](namehashing.md).
+Each ERC-721 token can be identified by a unique number, its `tokenId`. To make domains identifiable, we use a process called [Namehashing](namehashing.md).
 
-Here's an example of "example.crypto" namehash: `0xd584c5509c6788ad9d9491be8ba8b4422d05caf62674a98fbf8a9988eeadfb7e`
+For instance, `example.crypto`'s namehash:
+`0xd584c5509c6788ad9d9491be8ba8b4422d05caf62674a98fbf8a9988eeadfb7e`
 
 ### Visualization
 
-Here is a map to help visualize how the CNS Registry and Resolvers interact.
+This flow describes how the CNS `Registry` and `Resolvers` interact.
 
 ![](../.gitbook/assets/Smart-Contract-Architecture-Administration.svg)
 
 ### Registry
 
-Registry is the most essential smart contract in CNS. This is the contract that defines ownership rules, how domains are minted, provides [ERC-721 token metadata](https://docs.openzeppelin.com/contracts/2.x/api/token/erc721#IERC721Metadata), and stores a list of all domains with information about them.
+`Registry` is the most essential smart contract in CNS. This is the contract that defines ownership rules, how domains are minted, provides [ERC-721 token metadata](https://docs.openzeppelin.com/contracts/2.x/api/token/erc721#IERC721Metadata), and stores a metadata-enriched list of all domains.
 
-Registry stores the following data about each domain:
+`Registry` stores:
 
-* Owner address
-* Approved operator address
-* Resolver address
-* Domain name
+- Owner address
+- Approved operator address
+- Resolver address
+- Domain name
 
 {% hint style="info" %}
-As domains are identified by a namehash, we also store a domain name, apart from the other data. As it's impossible to calculate a domain name from its identifier, it allows users, that know only a namehash, lookup a real domain name.
+Since domains are identified by a namehash we also explicitly store the domain's name. This allows users who only know a namehash to look up a record by its name.
```

Comment:
> This allows users who only know a namehash to look up a record by its name

I wanted to stress that a user can look up a domain name by a namehash. Looking up a record by its (record's?) name, when you know only a namehash, shifts the emphasis and sounds a bit confusing to me.

Reply:
That makes sense, how about
> To make record retrieval easier we store the plaintext domain name on each record. This allows users to search by either the record identifier (namehash) or the domain name directly.
?

Reply:
Well, our smart contract API doesn't really allow to resolve a domain by its name. These details are probably too low level for this kind of article, but for our users, it would still involve looking up a namehash by a domain name and then resolving a domain by its namehash.

If you don't want to include such details here, I'm fine with the current version.

Reply:
No, that sounds like an important point. I'll add a clarification.